src/diagnostics/core_diagnostics.jl

```diff
@@ -857,6 +860,130 @@ add_diagnostic_variable!(
     compute! = compute_hussn!,
 )
 
+compute_cdnc!(out, state, cache, time) =
+    compute_cdnc!(out, state, cache, time, cache.atmos.precip_model)
+compute_cdnc!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("cdnc", model)
+
+function compute_cdnc!(
+    out,
+    state,
+    cache,
+    time,
+    precip_model::Microphysics2Moment,
+)
+    if isnothing(out)
+        return state.c.ρn_liq
+    else
+        out .= state.c.ρn_liq
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "cdnc",
+    long_name = "Cloud Liquid Droplet Number Concentration",
+    standard_name = "number_concentration_of_cloud_liquid_water_particles_in_air",
+    units = "m^-3",
+    comments = """
+    This is calculated as the number of cloud liquid water droplets in the grid 
+    cell divided by the cell volume.
+    """,
+    compute! = compute_cdnc!,
+)
+
+compute_cdinc!(out, state, cache, time) =
+    compute_cdinc!(out, state, cache, time, cache.atmos.precip_model)
+compute_cdinc!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("cdinc", model)
+
+function compute_cdinc!(
+    out,
+    state,
+    cache,
+    time,
+    precip_model::Microphysics2Moment,
+)
+    if isnothing(out)
+        return state.c.ρn_ice
+    else
+        out .= state.c.ρn_ice
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "cdinc",
+    long_name = "Cloud Ice Chrystal Number Concentration",
+    standard_name = "number_concentration_of_cloud_ice_chrystals_in_air",
+    units = "m^-3",
+    comments = """
+    This is calculated as the number of cloud ice chrystals in the grid 
+    cell divided by the cell volume.
+    """,
+    compute! = compute_cdinc!,
+)
+
+compute_ncra!(out, state, cache, time) =
+    compute_ncra!(out, state, cache, time, cache.atmos.precip_model)
+compute_ncra!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("ncra", model)
+
+function compute_ncra!(
+    out,
+    state,
+    cache,
+    time,
+    precip_model::Microphysics2Moment,
+)
+    if isnothing(out)
+        return state.c.ρn_rai
+    else
+        out .= state.c.ρn_rai
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "ncra",
+    long_name = "Raindrop Number Concentration",
+    standard_name = "number_concentration_of_raindrops_in_air",
+    units = "m^-3",
+    comments = """
+    This is calculated as the number of raindrops in the grid cell divided
+    by the cell volume.
+    """,
+    compute! = compute_ncra!,
+)
+
+compute_ncsn!(out, state, cache, time) =
+    compute_ncsn!(out, state, cache, time, cache.atmos.precip_model)
+compute_ncsn!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("ncsn", model)
+
+function compute_ncsn!(
+    out,
+    state,
+    cache,
+    time,
+    precip_model::Microphysics2Moment,
```

Comment:
Same as for cloud ice. Especially since P3 doesn't have two separate categories for them anyway